torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -540,11 +541,25 @@ class TORCH_CUDA_API Allocate : public Expr {
     return buffer_->getDataType().value();
   }
 
+  Allocate* alias() const {
+    return alias_;
+  }
+
+  void setAlias(Allocate* alias) {
+    TORCH_INTERNAL_ASSERT(alias->getMemoryType() == memory_type_);
+    // TODO: TORCH_INTERNAL_ASSERT(alias->size() == size_);
+    alias_ = alias;
+  }
+
  private:
   Val* buffer_ = nullptr;
   MemoryType memory_type_ = MemoryType::Local;
   Val* size_ = nullptr;
   bool zero_init_ = false;
+
+  // This alias tracks the next Allocate node in a linked chain of aliases
```

Comment:
Do we have any checks against cycles in this list?

Reply:
There isn't an explicit check against cycles. The CUDA kernel would fail in this case.